updated documentation of delete (#144)
cedardb/docs: content/references/sqlreference/statements/delete.md

```sql
delete from pending_notifications n
using channels c
where n.receiver_id = $1
  and n.channel_id = c.id
returning *;
```

> The returning clause refers to the state of the database after the delete is completed. Any deleted tuples are no longer visibles to scans, that may occur in the returning clause. 

Reviewer: The first sentence is a bit misleading because the deleted tuples are returned, how about just
```suggestion
Scans in the returning clause read the database after the delete is completed.
```

Author: That phrasing is significantly better!
